code/modules/admin/antagonist_panel/antagonist_panel_data.dm

```diff
@@ -0,0 +1,128 @@
+/**
+ *	The global singleton datum responsible for generating and providing data to individual instances of antagonist panels.
+ *	Such a datum is necessary due to the relatively large amount of antagonist data that is required to be processed with
+ *	each update of individual antagonist panels; tab data may be collated once every update then stored in a cache in order
+ *	to be fetched by antagonist panels.
+ */
+/datum/antagonist_panel_data
+	/// An associative list of tab indexes and their repective tab instances.
+	var/list/datum/antagonist_panel_tab/antagonist_panel_tabs
+	/// A list of tab names and their respective tab index.
+	var/list/list/antagonist_panel_tab_names
+	/// An associative list of tab indexes and their repective cached tab data.
+	var/list/list/ui_data_cache
+
+/datum/antagonist_panel_data/New()
+	. = ..()
+
+	src.antagonist_panel_tabs = list()
+	src.antagonist_panel_tab_names = list()
+	src.ui_data_cache = list()
+
+/// Returns the UI data for a specified antagonist panel tab, either fetching it from the cache or regenerating it depending on a cooldown.
+/datum/antagonist_panel_data/proc/request_ui_data(tab_index)
+	if (ON_COOLDOWN(global, "request_ui_data-[tab_index]", 0.9 SECONDS))
+		return src.ui_data_cache[tab_index]
+
+	src.get_antagonist_panel_tabs()
+
+	var/list/tab_section_data
+	var/list/antagonist_locations
+	var/list/mortality_rates
+
+	var/datum/antagonist_panel_tab/antagonist_tab = src.antagonist_panel_tabs[tab_index]
+	if (antagonist_tab)
+		tab_section_data = antagonist_tab.generate_section_data()
+		antagonist_locations = antagonist_tab.generate_location_data()
+
+	else
+		mortality_rates = src.generate_mortality_rates()
+
+	src.ui_data_cache[tab_index] = list(
+		"tabs" = src.antagonist_panel_tab_names,
+		"currentTabSections" = tab_section_data,
+		"mindLocations" = antagonist_locations,
+		"mortalityRates" = mortality_rates,
+	)
+
+	return src.ui_data_cache[tab_index]
+
+/// Populates `antagonist_panel_tabs` and `antagonist_panel_tab_names` with the necessary antagonist tab types and names.
+/datum/antagonist_panel_data/proc/get_antagonist_panel_tabs()
+	if (ON_COOLDOWN(global, "get_antagonist_panel_tabs", 0.85 SECONDS))
```

Comment:
why time

Reply:
0.9 seconds is the established duration of a TGUI process tick, however as that time is subject to deviation, a cooldown of 0.85 seconds ensures that `get_antagonist_panel_tabs()` will run once every tick.

Reply:
could use a comment or define to that effect